src/components/AirbaseNG/AirbaseNG.tsx

```diff
@@ -1,165 +1,171 @@
+            replayInterface: "wlan0", // Default interface
             macAddress: "",
             filePath: "",
+            verbosity: false,
+            disableBeacons: false,
             customConfig: "",
+            customScript: "",
+        },
+
+        // Validation: Ensure essential fields are filled.
+        validate: {
+            fakeHost: (value) => (value ? null : "Fake host name is required."),
+            channel: (value) => (value ? null : "Channel is required."),
+            replayInterface: (value) => (value ? null : "WLAN interface is required."),
         },
     });
 
-    // Check if the command is available and set the state variables accordingly.
+    // Effect to check command availability on component mount.
     useEffect(() => {
-        // Check if the command is available and set the state variables accordingly.
         checkAllCommandsAvailability(dependencies)
             .then((isAvailable) => {
-                setIsCommandAvailable(isAvailable); // Set the command availability state
-                setOpened(!isAvailable); // Set the modal state to opened if the command is not available
-                setLoadingModal(false); // Set loading to false after the check is done
+                setIsCommandAvailable(isAvailable);
+                setOpened(!isAvailable);
+                setLoadingModal(false);
             })
-            .catch((error) => {
+            .catch((error: any) => {
+                // Fix the error type
                 console.error("An error occurred:", error);
-                setLoadingModal(false); // Also set loading to false in case of error
+                setLoadingModal(false);
             });
     }, []);
 
     /**
-     * handleProcessData: Callback to handle and append new data from the child process to the output.
-     * It updates the state by appending the new data received to the existing output.
-     * @param {string} data - The data received from the child process.
+     * Handles incoming data from the child process.
+     * @param {string} data - Data received from the process.
      */
     const handleProcessData = useCallback((data: string) => {
-        setOutput((prevOutput) => prevOutput + "\n" + data); // Append new data to the previous output.
+        setOutput((prevOutput) => prevOutput + "\n" + data);
     }, []);
 
     /**
-     * handleProcessTermination: Callback to handle the termination of the child process.
-     * Once the process termination is handled, it clears the process PID reference and
-     * deactivates the loading overlay.
-     * @param {object} param - An object containing information about the process termination.
-     * @param {number} param.code - The exit code of the terminated process.
-     * @param {number} param.signal - The signal code indicating how the process was terminated.
+     * Handles the termination of the child process.
+     * @param {object} param - Process termination details.
```

Comment:
Keep the original comment. It is more descriptive. 
```suggestion
   * @param {object} param - An object containing information about the process termination.
```